### PR TITLE
fix: list pipeline template versions

### PR DESCRIPTION
**Pull request by pritamstyz4ever**
## Context

fix the issue with listing pipeline template versions

## Objective

fix the issue with listing pipeline template versions

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by minghay — APPROVED** [review]

**Comment by sagar1312** [comment]
What was the issue?

File: lib/pipelineTemplateVersionFactory.js [hunk 1]
```diff
@@ -162,9 +162,12 @@ class PipelineTemplateVersionFactory extends BaseFactory {
...
         }
 
-        config.params.templateId = pipelineTemplateMeta.id;
-
-        return super.list(config);
+        return super.list({
+            ...config,
+            params: {
+                templateId: pipelineTemplateMeta.id
+            }
```
Comment (sagar1312) [inline lib/pipelineTemplateVersionFactory.js]: We should not override the params with just `templateId`. There could be other params (Ex: `createTime`) besides `name` and `namesapce`.
Instead, exclude `name` and `namespace` from the params.

**Review by sagar1312 — CHANGES_REQUESTED** [review]

**Review by tkyi — APPROVED** [review]

File: lib/pipelineTemplateVersionFactory.js (hunk 1 above)
Comment (pritamstyz4ever) [reply lib/pipelineTemplateVersionFactory.js]: not really, `pipelineTemplateVersion` schema does not have name,namespace, only way to list is via templateId

Comment (sagar1312) [reply lib/pipelineTemplateVersionFactory.js]: I agree that we don't need to exclude `name` and `namespace` as these get ignored while constructing DB query. But, it is still unclear on why we are picking only `templateId` from params.
Model interface should not be designed/restricted due to current use cases in the Screwdriver API. It should allow filtering on any of the `pipelineTemplateVersion` field.

Comment (pritamstyz4ever) [reply lib/pipelineTemplateVersionFactory.js]: the params don't get ignored I think, so we need to remove them.
https://github.com/screwdriver-cd/models/blob/master/lib/baseFactory.js#L132

Comment (sagar1312) [reply lib/pipelineTemplateVersionFactory.js]: Yes, looks like if there are params not specified in the schema, error is thrown
https://github.com/screwdriver-cd/datastore-sequelize/blob/master/index.js#L475

Then, as originally suggested, let us exclude `name` and `namespace` from `config.params`.

Or, if we want to be strict and reserve `config.params` to only allow the fields of `pipelineTemplateVersion`, then should we have `name` and `namespace` on `config` and not in `config.params`?

Comment (pritamstyz4ever) [reply lib/pipelineTemplateVersionFactory.js]: Yes, I think we can move `name` and `namespace` from `config.params`

File: lib/pipelineTemplateVersionFactory.js [hunk 2]
```diff
@@ -149,19 +150,22 @@ class PipelineTemplateVersionFactory extends BaseFactory {
             config.params = {};
         }
 
-        if (!config.params.name || !config.params.namespace) {
+        if (config.params.templateId) {
+            return super.list(config);
+        }
```
Comment (sagar1312) [inline lib/pipelineTemplateVersionFactory.js:155]: When `templateId` params exist, why are we not considering `name` and `namespace`?
When `templateId`, `name` and `namespace` are specified, we should return versions associated with the template matching `templateId`, `name` and `namespace`.

**Review by sagar1312 — CHANGES_REQUESTED** [review]

**Review by sagar1312 — APPROVED** [review]

**Pull request merged**